pkg/apis/azureprovider/v1beta1/types.go

```diff
@@ -401,11 +401,17 @@ type Image struct {
 type VMIdentity string
 
 type OSDisk struct {
-	OSType      string      `json:"osType"`
-	ManagedDisk ManagedDisk `json:"managedDisk"`
-	DiskSizeGB  int32       `json:"diskSizeGB"`
+	OSType      string                `json:"osType"`
+	ManagedDisk ManagedDiskParameters `json:"managedDisk"`
+	DiskSizeGB  int32                 `json:"diskSizeGB"`
 }
 
-type ManagedDisk struct {
-	StorageAccountType string `json:"storageAccountType"`
+type ManagedDiskParameters struct {
+	StorageAccountType string                       `json:"storageAccountType"`
+	DiskEncryptionSet  *DiskEncryptionSetParameters `json:"diskEncryptionSet,omitempty"`
+	ID                 string                       `json:"id,omitempty"`
+}
+
+type DiskEncryptionSetParameters struct {
+	ID string `json:"id,omitempty"`
```

Comment:
Would be nice to include some documentation on the the expected form of the ID, Using Azure resource IDs implies that this resource can be from any subscription, any resource group.

Since this is for supporting customer managed encryption, it is fair to say we need to support this resource from resource group other than the cluster's but what about the subscription.

Reply:
Here is an example of an a disk encryption set ID.  Azure resource IDs include a path-like system so that it is clear where the object is located, what the object is, and its name.
```
/subscriptions/<subscriptionid>/resourceGroups/disk-encryption/providers/Microsoft.Compute/diskEncryptionSets/aro-de
```
According to the documentation here https://docs.microsoft.com/en-us/azure/virtual-machines/linux/disk-encryption-key-vault#create-a-key-vault it states:
```
Your key vault and VMs must be in the same subscription. Also, to ensure that encryption secrets don't cross regional boundaries, Azure Disk Encryption requires the Key Vault and the VMs to be co-located in the same region. Create and use a Key Vault that is in the same subscription and region as the VMs to be encrypted.
```